src/protections/PropagateRoomBan.ts

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2021 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { StringProtectionSetting } from './ProtectionSettings';
+import { Mjolnir } from '../Mjolnir';
+import { recommendationToStable, RECOMMENDATION_BAN } from '../models/ListRule';
+import { logMessage } from '../LogProxy';
+import { LogLevel } from 'matrix-bot-sdk';
+import { RULE_USER } from '../models/BanList';
+import { Protection } from './IProtection';
+
+export class PropagateRoomBan extends Protection {
+    settings = {
+        banlistShortcode: new StringProtectionSetting(),
```

Comment:
internally we use `BanList` and `banList` so this should probably be `banListShortcode`

Reply:
Do you think that this should be tied to to the default list used by Mjolnir's own ban command, to avoid confusion? 

(Like this https://github.com/matrix-org/mjolnir/blob/v1.3.2/src/commands/UnbanBanCommand.ts#L33-L42 )

Reply:
Probably we should default to the default list, but I think we should allow the admins to overwrite it.

Reply:
So I made a change where it defaults to the default banlist, but still allows it to be overwritten. Tell me what you think. The main reason I want it to be overwritable because I would like to separate the room admin bans from the server admin bans (since the latter have a higher trust level).

Reply:
>Probably we should default to the default list, but I think we should allow the admins to overwrite it

That seems fair and the way you have done this works. :+1: